lib/post/pages/create_comment_page.dart

```diff
@@ -0,0 +1,355 @@
+    } else if (widget.commentView != null) {
+      replyingToAuthor = widget.commentView?.commentView?.creator.name;
+      replyingToContent = widget.commentView?.commentView?.comment.content;
+    } else if (widget.comment != null) {
+      replyingToAuthor = widget.parentCommentAuthor;
+      replyingToContent = widget.comment?.content;
+    }
+
+    _bodyTextController.addListener(() {
+      WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+        setState(() => isSubmitButtonDisabled = _bodyTextController.text.isEmpty);
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    _bodyTextController.dispose();
+    _bodyFocusNode.dispose();
+
+    FocusManager.instance.primaryFocus?.unfocus();
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final ThunderState thunderState = context.read<ThunderBloc>().state;
+
+    return MultiBlocListener(
+      listeners: [
+        BlocListener<PostBloc, PostState>(
+          listenWhen: (previous, current) {
+            return previous.status != current.status;
+          },
+          listener: (listenerContext, state) {
+            if (state.status == PostStatus.success) {
+              if (hasExited == false) Navigator.pop(context);
+              setState(() => hasExited = true);
+            }
+            if (state.status == PostStatus.loading || state.status == PostStatus.refreshing) {
+              setState(() {
+                isLoading = true;
+              });
+            }
+            if (state.status == PostStatus.failure) {
+              setState(() {
+                isLoading = false;
+              });
+              ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                content: Text(state.errorMessage ?? AppLocalizations.of(context)!.unexpectedError),
+                duration: const Duration(days: 1),
+              ));
+            }
+          },
+        ),
+        if (widget.comment != null)
+          BlocListener<InboxBloc, InboxState>(
+            listenWhen: (previous, current) {
+              return previous.status != current.status;
+            },
+            listener: (listenerContext, state) {
+              if (state.status == InboxStatus.success) {
+                if (hasExited == false) Navigator.pop(context);
+                setState(() => hasExited = true);
+              }
+              if (state.status == InboxStatus.loading || state.status == InboxStatus.refreshing) {
+                setState(() {
+                  isLoading = true;
+                });
+              }
+              if (state.status == InboxStatus.failure) {
+                setState(() {
+                  isLoading = false;
+                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                    content: Text(AppLocalizations.of(context)!.unexpectedError),
+                    duration: const Duration(days: 1),
```

Comment:
This snackbar will appear for a whole day?? 

Also a quick note about all snackbars, I'm working on a refactor in #571, so depending on who goes first, I'll include this in my changes or vice versa!

Reply:
Right, there I was just trying to make it stay indefinitely until the user gets rid of it. I guess it might not be needed. Should I just remove the duration? I am also aware of the refactor, I can change my snackbars once that is merged.

Reply:
Ahh I see what you're going for. Currently all of the snackbars auto-dismiss, so this would be a new pattern. I might not even think to dismiss it unless it had an `x` button. But of course I'm totally open to a new approach. I'd be interested to hear from others.

Reply:
The refactor should now be merged in! I think we can just set a long duration on the snackbars for now - enough time so that the message can be read or a screenshot can be taken